torch/csrc/autograd/utils/lambda_post_hook.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <torch/csrc/autograd/function_hook.h>
+
+namespace torch {
+namespace autograd {
+namespace utils {
+
+// Turns lambda into a torch::autograd::FunctionPostHook.
+class LambdaPostHook : public torch::autograd::FunctionPostHook {
+  using variable_list = std::vector<torch::autograd::Variable>;
+
+ public:
+  /* implicit */ LambdaPostHook(std::function<void(const variable_list&)> fn)
+      : fn_(std::move(fn)) {}
+
+  variable_list operator()(
+      const variable_list& outputs,
+      const variable_list& inputs) override {
+    fn_(inputs);
```

Comment:
I didn't read through what the lambda_post_hook is for, but is this `lambda_post_hook` only supposed to be used by AccmulateGrad? It seems that we are always passing the inputs in, although it is ok since other variables can be passed in by capture.

Reply:
Currently, it is only used for AccumulateGrad.